ch_pipeline/processing/quarterstack.py

```diff
@@ -0,0 +1,332 @@
+    the database, will not effect the stacks to be generated.
+    """
+
+    type_name = "quarterstack"
+    tag_pattern = r"(?P<year>\d{4})q(?P<quarter>[1-4])p(?P<partition>\d)"
+
+    # Parameters of the job processing
+    default_params = {
+        # Daily processing revisions to use (later entries in this list take precedence
+        # over earlier ones)
+        "daily_revisions": ["rev_02"],
+        "daily_root": "/project/rpp-krs/chime/chime_processed/",
+        # Frequencies to process
+        "freq": [0, 1024],
+        # The beam transfers to use (need to have the same freq range as above)
+        "product_path": "/project/rpp-krs/chime/bt_empty/chime_4cyl_allfreq/",
+        "partitions": 2,
+        # Don't generate quarter stacks with less days than this
+        "min_days": 5,
+        # RA range used to estimate the cross talk for each quarter
+        "crosstalk_ra": {
+            "q1": [165, 180],
+            "q2": [240, 255],
+            "q3": [315, 330],
+            "q4": [45, 60],
+        },
+        # Job params
+        "time": 180,  # How long in minutes?
+        "nodes": 16,  # Number of nodes to use.
+        "ompnum": 6,  # Number of OpenMP threads
+        "pernode": 8,  # Jobs per node
+    }
+
+    default_script = DEFAULT_SCRIPT
+
+    def _create_hook(self):
+        """Create the revision.
+
+        This tries to determine which days are good and bad, and partitions the
+        available good days into the individual stacks.
+        """
+
+        days = {}
+
+        core.connect()
+
+        # Go over each revision and construct the set of LSDs we should stack, and save
+        # the path to each.
+        # NOTE: later entries is `daily_revisions` will override LSDs found in earlier
+        # revisions.
+        for rev in self.default_params["daily_revisions"]:
+
+            daily_path = (
+                self.root_path
+                if self.default_params["daily_root"] is None
+                else self.default_params["daily_root"]
+            )
+            daily_rev = daily.DailyProcessing(rev, root_path=daily_path)
+
+            # Get all the bad days in this revision
+            revision = df.DataRevision.get(name=rev)
+            query = (
+                df.DataFlagOpinion.select(df.DataFlagOpinion.lsd)
+                .distinct()
+                .where(
+                    df.DataFlagOpinion.revision == revision,
+                    df.DataFlagOpinion.decision == "bad",
+                )
+            )
+            bad_days = [x[0] for x in query.tuples()]
+
+            # Get all the good days
+            query = (
+                df.DataFlagOpinion.select(df.DataFlagOpinion.lsd)
+                .distinct()
+                .where(
+                    df.DataFlagOpinion.revision == revision,
+                    df.DataFlagOpinion.decision == "good",
+                )
+            )
+            good_days = [x[0] for x in query.tuples()]
```

Comment:
What's the rest of the tuple? Can this be done with a `.get()` or similar instead?

Reply:
I think this is about the most minimal query you can do, the `.tuples()` call means it's returning a tuple of all the fields specified within the `.select(...)` for each row. As I only specify one field I need to pull it out of each 1-tuple. I don't think there's a cleaner way to do this.